src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -218,10 +219,16 @@ private void RemoveProcessedTransactions(IReadOnlyList<Transaction> blockTransac
                 {
                     discoveredForPendingTxs++;
                 }
+
+                if (blockTransactions[i].IsEip1559)
+                {
+                    eip1559Txs++;
+                }
             }
             
             Metrics.DarkPoolRatioLevel1 = transactionsInBlock == 0 ? 0 : (float)discoveredForHashCache / transactionsInBlock;
             Metrics.DarkPoolRatioLevel2 = transactionsInBlock == 0 ? 0 : (float)discoveredForPendingTxs / transactionsInBlock;
+            Metrics.Eip1559TransactionsRatio = eip1559Txs == 0 ? 0 : (float)eip1559Txs / transactionsInBlock;
```

Comment:
Maybe on empty blocks those should be 1 not 0?

Reply:
Also you are checking eip1559Txs for 0, while you should transactionsInBlock , so you can have division by 0 here

Reply:
Can you change it to if...else... not to do same check 3 times? Less bug-prone

Reply:
oh, thanks for catching! I think 1 can be misleading too... I think we can assign new value only if block isn't empty and on empty block do not assign anything and just leave previous value